### ISSUE TITLE
iOS and Dropbox synchronisation not working

### ISSUE THREAD

**Issue by akhiar** [laurent22/joplin#6375]
## Environment

Joplin version: 12.7.1 (iOS App)
Platform: iOS
OS specifics: iOS 15.4.1

## Steps to reproduce

1. Install the app
2. Configuration > Sync target = Dropbox
3. Link with Dropbox account
4. Synchronise (ISSUE: No errors but notes are not synced. Log message "Network request failed")

## Describe what you expected to happen
PC app works just fine. Only happens in iOS app.

## Logfile

![20220407_041606346_iOS](https://user-images.githubusercontent.com/3339621/162121460-7ff62d56-3852-45a7-a541-79fd7967b3d6.png)

**Comment by lab6zine** [comment]
I am also affected by this bug.

I note that this has started to happen within the last 48-72 hours, having worked fine previously.

**Comment by bellrichm** [comment]
I also am recently seeing this. I have no firm evidence, but it seems to have started after I updated the Dropbox app on my iOS devices.

**Comment by sif** [comment]
Seeing this as well. I didn't do anything different except perhaps a power outage the night before.

**Comment by sif** [comment]
Seeing this as well. I didn't do anything different except perhaps a power outage the night before. I will note that it works for one of my iOS device.

**Comment by cspenn** [comment]
Chiming in, iOS app, Monterey desktop. Desktop to desktop on different machines works fine, it's just iOS. Deleted Joplin app and reinstalled from app store, still having the issue.

**Comment by sif** [comment]
I made a quick check on Dropbox's status page (https://status.dropbox.com/) and for the days Joplin was out for me, Dropbox was up.

I was able to intercept the calls made by my Joplin (thank you roman_r_m) and was seeing this error:
"Certificate verify failed: self signed certificate"

I was also seeing this repeated:
"Client TLS handshake failed. The client disconnected during the handshake. If this happens consistently for [service], this may indicate that the client does not trust the proxy's certificate."

The part where it says [service] includes joplinapp.org, *.googleapis.com, content.dropboxapi.com, etc.

I'm going to assume I setup mitmproxy incorrectly and welcome additional investigation into this.

**Comment by sif** [comment]
I have also tested OneDrive and sync works. Switching back to Dropbox on the same device did not fix this issue.

**Comment by seanieb** [comment]
Dropbox sync appears to be broken for me on iOS too. It works perfectly from my MacOS desktop to my MacOS desktop using Dropbox Sync.

**Comment by seanieb** [comment]
@sif The "Certificate verify failed: self signed certificate" was probably due to the local cert you were using to intercept the traffic. I haven't seen any issues with the cert. You can see the cert here:
https://www.ssllabs.com/ssltest/analyze.html?d=content.dropboxapi.com

**Comment by sif** [comment]
I was using the cert from here: https://mitm.it

**Comment by roman-r-m** [comment]
> I was using the cert from here: https://mitm.it

FWIW When I open the link in Firefox I'm getting this:

> Firefox does not trust this site because it uses a certificate that is not valid for mitm.it. The certificate is only valid for the following names: cloudfront.net, *.cloudfront.net

Though not sure if it's the same cert you have installed.

UPD
Probably a different certificate, ignore me.

**Comment by csilker** [comment]
My Dropbox sync is also broken since updating the Dropbox app two days ago on my iPad Pro (15.4.1). It's syncing fine in Windows and Android.

I uninstalled Joplin and reinstalled it, including re-authorizing it in Dropbox. Everything syncs and then a bit later it stops. The app looks like it's syncing, syncs for 23s, and doesn't actually sync. The error from the log is attached.
![F3600757-5FA4-489A-80F8-DD31C6FB865F](https://user-images.githubusercontent.com/14827297/162524804-8768df27-fc50-4ab8-a60e-542c773465aa.png)

**Comment by sif** [comment]
On iOS, I don't think updating Dropbox should affect Joplin. It's my understanding that Apple sandboxes third party apps: [Security of runtime process in iOS and iPadOS](https://support.apple.com/guide/security/security-of-runtime-process-sec15bfe098e/web)
> "All third-party apps are “sandboxed,” so they are restricted from accessing files stored by other apps or from making changes to the device."

**Comment by csilker** [comment]
@sif Could this be from updating to 15.4.1? That happened slightly before the Dropbox update.

**Comment by sif** [comment]
The main content of iOS 15.4.1 is "AppleAVD." It was released at the end of March. This is Apple's audio and video decoder framework. Part of what makes it a powerful exploit is that it runs in kernel space.

Dropbox's latest version is 274.2 and was released on April 5th. The previous version (272.2) was released on March 22nd. Dropbox puts this in their changelog for the latest:
> • You can now easily share and star your photos and files with one click in the preview screen.
> • We fixed an issue that was causing the Dropbox app to consume a lot of battery. The app will now use significantly less battery and run much more efficiently!

I doubt Joplin is affected by either changes. However, the timing of user reports makes me suspect Dropbox more than Apple. That is, Dropbox may have done something to their APIs and that affects how Joplin makes the calls.

**Comment by natecrow** [comment]
I'm getting this error as well on my iPhone, which is on iOS 15.3.1, and my iPad, which is on iOS 15.1, so I don't think the error is related to iOS 15.4.1. Also, I don't have the Dropbox app installed on either device, although I am syncing to Dropbox.

**Comment by rtoip** [comment]
> I am also affected by this bug.

I can confirm that I can reproduce synchronization problem on two of my iOS devices (15.3.1, 15.3). First observation of the bug occurred ~ 24-36 hours ago.

A Windows 10 based Joplin installation works with Dropbox.

**Comment by laurent22** [comment]
Anyone managed to trace the requests the app is making (with Wireshark or similar) to see what call specifically is failing? Also can someone replicate this by building the app and running it in the iOS simulator?

**Comment by seanieb** [comment]
@laurent22 I spoke to a dev at Dropbox to see what they can see from my linked app on their side.

> I do see a `not_found` on a download call for you for that app, but that is supposed to happen if the `path` is wrong. I can’t see the actual `path` value there though, and the app would hopefully be able to handle that sort of thing anyway, since pretty much all Dropbox API endpoints can return errors like that in certain conditions.

> I can repro the error in the log, but it’s not too much help since it doesn’t actually show what it’s doing (and likewise I can’t see actual parameters in exclog). ¯\_( ˘͡ ˘̯)_/¯

> So, yea, dev should repro in dev environment and see what’s up. If the API is bungling something they can report it to us of course, but unfortunately I can’t see it from this.

**Comment by solace** [comment]
I am getting the `Network request failed` error, can't do the build and sim at test at this time, but at some point prior to those errors, I see:

“remote does not exist, local is new and has never been synced”

![image](https://user-images.githubusercontent.com/374583/162607263-5441aeca-01d6-4a51-91c9-0f0e798bf7c3.png)

Not sure if that is related to the issue or not.

I've tried deleting the iOS app and syncing it from scratch and it will do the initial sync, and I am certain it does work for a very short amount of time afterwards (ie. update and sync on desktop connected via filesystem to the Dropbox folder then manually trigger sync on iOS). But after a time, it just stops again. Says sync is complete with no changes, and the logs show the network error.

The remote DNE message popped up at some point after the reinstall but I cannot confirm if that directly coincided with when it stopped working.

**Comment by cjdaniel** [comment]
Could this error have something to do with iOS' strict requirements for SSL connections?

I remember seeing a similar error when I was using Nextcloud with a Let's Encrypt cert as a sync target. I gave up and moved to Dropbox. Now I am seeing this error as well.

**Comment by sif** [comment]
> Could this error have something to do with iOS' strict requirements for SSL connections?

If I remember correctly, Apple has required SSL [since 2016](https://techcrunch.com/2016/06/14/apple-will-require-https-connections-for-ios-apps-by-the-end-of-2016/). Joplin has been providing that for as long as I remember and when I traced it with Wireshark yesterday, I did see evidence of SSL activities. So I doubt this is the issue.

**Comment by sif** [comment]
> Could this error have something to do with iOS' strict requirements for SSL connections?

If I remember correctly, Apple has required SSL [since 2017](https://techcrunch.com/2016/06/14/apple-will-require-https-connections-for-ios-apps-by-the-end-of-2016/). Joplin has been providing that for as long as I remember and when I traced it with Wireshark yesterday, I did see evidence of SSL activities. So I doubt this is the issue.

**Comment by laurent22** [comment]
> Could this error have something to do with iOS' strict requirements for SSL connections?

Or maybe Dropbox TLS certificate is somehow special. Anyone tried to file a report on their forum?

**Comment by seanieb** [comment]
Their cert is not new, been there for a few months.
[Dropbox SSL cert ](https://www.ssllabs.com/ssltest/analyze.html?d=content.dropboxapi.com&latest)

What else could have changed a few days ago?

**Comment by cjdaniel** [comment]
Available/required ciphers could have changed, perhaps. That is a web server/proxy configuration.

**Comment by seanieb** [comment]
"Reminder: The Dropbox API will no longer accept TLS 1.0 or 1.1 "
However, the deadline for this isn't for two days. I'll ask and see if they made the change earlier. 

https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Reminder-The-Dropbox-API-will-no-longer-accept-TLS-1-0-or-1-1/td-p/582785

**Comment by cjdaniel** [comment]
> "Reminder: The Dropbox API will no longer accept TLS 1.0 or 1.1 "
> However, the deadline for this isn't for two days. I'll ask and see if they made the change earlier.

I am not sure if this is the hostname being connected to, but a [link to an SSL test was shared earlier](https://www.ssllabs.com/ssltest/analyze.html?d=content.dropboxapi.com). This host supports TLS 1.0 and 1.1.

**Comment by seanieb** [comment]
> I am not sure if this is the hostname being connected to, but a [link to an SSL test was shared earlier](https://www.ssllabs.com/ssltest/analyze.html?d=content.dropboxapi.com). This host supports TLS 1.0 and 1.1.

Same for `api.dropbox.com`
https://www.ssllabs.com/ssltest/analyze.html?d=api.dropboxapi.com&s=162.125.1.19&latest

Very odd. What else could change?

**Comment by laurent22** [comment]
> Very odd. What else could change?

Maybe a root certificate they use and that has expired recently? That's what happened with Let's Encrypt.

**Comment by seanieb** [comment]
> Maybe a root certificate they use and that has expired recently? That's what happened with Let's Encrypt.

The Root cert is DigiCert, and it looks good. A bad root cert would impact a lot more apps that use their API, not just the joplin-ios app.

**Comment by roman-r-m** [comment]
> Maybe a root certificate they use and that has expired recently? That's what happened with Let's Encrypt.

This would affect all platforms, yet the issue seems to be limited to iOS only.

**Comment by laurent22** [comment]
If all platforms have an up-to-date certificate but not the iOS version for some reason it could happen.

For Let's Encrypt all platforms now have the right certificate expect old versions of Android. But I agree it would weird if that's what it was, but the whole problem is weird anyway.

**Comment by laurent22** [comment]
I'm wondering, are there any other React Native projects that use Dropbox? If there is perhaps they're having the same problem and may have a solution. I'm not on my laptop now, but if someone could check it could be worth it.

**Comment by soundalchemist** [comment]
Oof, I just spent 20 minutes fooling around with this problem on my phone. I'm actually relieved it's a bug and not something wrong with my config.

Joplin's sync isn't perfect. It is quirky, but I've gotten used to it being reliably quirky. This is the first real problem I've had in two years of use. Cheers to the devs!

**Comment by laurent22** [comment]
I wonder if it's related to this problem:

<https://www.dropboxforum.com/t5/Dropbox-files-folders/sharing-by-link-isn-t-working/td-p/589346>

It seems since a few days many people on the forum are saying they can't share files from iOS anymore.

**Comment by clintconklin** [comment]
Chiming in on this as well. 

Been having the same 'network request failed' issue syncing joplin on both my iphone and ipad with dropbox since upgrading to ios 15.4.1.

**Comment by inll5ge0u1v** [comment]
2 data points. A friend with Joplin 12.7.1 on iOS 15.3.1 ipad air 4th gen reports dropbox sync is fine and working as expected.

I am on Joplin 12.7.1 on ios 15.4.1 on iphone SE and ipad mini 5th gen and dropbox sync stopped working 4/7 21:55 (US eastern). The logs say the last successful sync was 4/7 21:52 (maybe? not sure if I'm reading the logs correctly).

additional notes: my friend from above upgraded ios, and sync still works. Friend has 1.43 GB, I have 392MB. I've also attached a screenshot of the first time I see this message . I think I did a manual sync, updated a note, and then another manual sync. 

![first error](https://user-images.githubusercontent.com/103471017/162949308-2bbc20f2-b14e-425b-9199-4d39da6fccfc.png)

**Comment by adcousin** [comment]
Issue encountered with Joplin 12.7.1, iOS 14.2, iPhone 6S.
First trace of the `Network request failed` on 04/07 11h04 (UTC+2)
Cheers and good luck in your investigations.

**Comment by aciba90** [comment]
I have the same problem with Joplin 12.7.1, iOS 15.4.1, iPhone 8.
Interestingly it works fine with and iPad (5th generation) and the same Joplin's and iOS' versions.

**Comment by n00b12345** [comment]
Same issue. iOS sync has completely stopped working.

**Comment by CarlosZBent** [comment]
I've been having the same issue for about a week. Error is `network request failed`. I have no issue on Windows (10 pro 1903) and Android (11), just iPadOS 15.4, on an iPad Air 2

**Comment by gavinr** [comment]
Is the only current workaround to switch sync providers?

**Comment by sif** [comment]
Gavin, if you don't have a lot of notes, you can move it to another provider. I don't think it would work if your Dropbox is bigger than the other providers and you need that much space.

**Comment by laurent22** [comment]
Anyone tried downgrading iOS to an earlier version? I can't even get the iOS simulator 15.4 so it might take a while before this issue is fixed.

**Comment by Daeraxa** [comment]
There are reports of it on 15.3, 15.1, 14.8 and 14.2 so doesn't seem to be a particular new version of iOS.

**Comment by laurent22** [comment]
> There are reports of it on 15.3, 15.1, 14.8 and 14.2 so doesn't seem to be a particular new version of iOS.

Yes but it seems to be happening more consistently on 15.4. I was half assuming that every "Network request failed" errors end up in this thread, even those not related to this new issue but I could be wrong.

**Comment by Daeraxa** [comment]
Fair enough, I have a work phone on 15.3 if you think that would be worth anything for a test.

**Comment by laurent22** [comment]
Unfortunately it's not possible to test just with the phone. I need to run the dev app in a simulator so that I can randomly change the request headers or parameters until it works.

**Comment by cajhin** [comment]
another data point: I installed Joplin for the first time ever, today

iOS *did* sync via dropbox, and downloaded the mac's desktop manual and a test note.
An hour later, it stopped working.
I can only guess:
- maybe the manual dropbox login that created the initial token was valid for an hour
- maybe related to the 3-connection limit that dropbox has for free accounts

update: uninstalled + reinstalled Joplin, logged into dropbox, sync works

**Comment by Daeraxa** [comment]
If it was related to the connection limits or token timeouts then surely we would be seeing this on desktop and android clients? Or does RN via iOS use a different method of accessing the API?

**Comment by laurent22** [comment]
Right, we should also see the same problem on desktop because the lib is the same. It could be that Dropbox is sending back new response headers and RN doesn't like some of them.

**Comment by sif** [comment]
The 3-connection limit was my initial theory too and if true, would be very disappointing. It is possible that this is a multi-layered problem.

**Comment by laurent22** [comment]
Could you try to disconnect one of the client to see if it helps? You can find the list of connected apps there: https://www.dropbox.com/account/connected_apps

**Comment by cajhin** [comment]
I had kicked out all apps and devices already, and it didn't help, at least not immediately. I suspect you need to explicitly sign in again with dropbox, which is what the re-install achieved (this just my best guess)

**Comment by soundalchemist** [comment]
I pay for a dropbox subscription, so there are no device limits, and I still encountered this bug.

**Comment by Daeraxa** [comment]
So I've never set up my dropbox for sync before. I set up the desktop app then the ios app (the dropbox website now shows a single Joplin connected app - 2 connected apps total (MS office is the other).
Deskop sync'd. Then iOS sync'd (it had a couple of extra notes on it so it created 9 remote files).
The very next sync I attempted failed with the same error being reported here.

~~Out of curiousity I tried disconnecting Joplin from connected apps and realised that is a bit of a pain to get configured again if you do that as Joplin doesn't allow you to do a first time Dropbox connection again.~~ After I waited a few mins it seemed to prompt for a new auth

**Comment by sif** [comment]
Both sync failed or just iOS?

**Comment by inll5ge0u1v** [comment]
I only have Joplin as a connected app to dropbox ( I do not use the desktop nor mobile dropbox client). I do have multiple devices with joplin , but joplin only shows up once in the connected app page.

I uninstalled joplin on my iphone, and my ipad still does not sync. I re-authed the connection on my iphone and 1 app is listed as connected. The initial sync seems to work: aside from Revision, the sync status on iphone matches the sync status on windows. I did get "too many requests" failures, and had to crank down the concurrent connections to 5 to complete the initial sync. Unfortunately, the next sync fails with "TypeError: Network request fail" message again. Setting concurrent connections to 1 did not help.

**Comment by Daeraxa** [comment]
> Both sync failed or just iOS?

Only iOS.

Also once auth access was granted again it still failed on iOS.

**Comment by pocph** [comment]
One other Data point.  I do most of my editing on PC and sync to Dropbox.  I have a Iphone and Android device that both sync to the same data.   Only the Iphone is having issues.

**Comment by laurent22** [comment]
I've installed macOS 12 and downloading latest XCode now which can go up to iOS simulator 15.2. I'm not optimistic I'll manage to replicate the bug on dev environment, but there's not much else to try.

**Comment by pocph** [comment]
Laurent - Not sure if this applies to anyone else but when I sync Joplin on the Iphone I'm usually including a picture as an attachment.    What else can I try to help ?

**Comment by Geraldhub** [comment]
Thanks, fix found. Hope this quickly gets into the apple store.

**Comment by soundalchemist** [comment]
Thank you! I'm looking at the piece of code you edited and it doesn't look fun!

**Comment by CarlosZBent** [comment]
Nice find, what a weird incompatibility. I guess it should be dealt with the Apple side of the error now, because the Dropbox recommended method is still POST, so it’s risky to leave it like that for long IMO.

**Comment by laurent22** [comment]
Someone reported on the forum that the latest iOS version solves the issue, so please give it a try.

**Comment by bellrichm** [comment]
> Someone reported on the forum that the latest iOS version solves the issue, so please give it a try.

Yes, it fixed it for me. Thanks!

**Comment by CarlosZBent** [comment]
> Someone reported on the forum that the latest iOS version solves the issue, so please give it a try.

it worked for me once after the update, it synchronized all the notes it had lagged behind my other devices. Now it's doing the same thing as before `Synchronizer: TypeError: Network request failed`. 
It's iPadOS 15.4 on an iPad Air 2.

**Comment by laurent22** [comment]
Even if you try to sync again? That's annoying. I wonder if certain other requests still result in an empty POST body. Is there any additional info before or after this error in the log?

**Comment by sif** [comment]
@CarlosZBent The latest version of iPadOS is 15.4.1. Can you try updating to that?

**Comment by CarlosZBent** [comment]
> @CarlosZBent The latest version of iPadOS is 15.4.1. Can you try updating to that?

I'll try that

**Comment by CarlosZBent** [comment]
> Even if you try to sync again? That's annoying. I wonder if certain other requests still result in an empty POST body. Is there any additional info before or after this error in the log?

At the time I updated, I did retry like 20 times, and it was giving me the mentioned error. Now, a few hours later, I retried with various operations and its working fine. My connection was ok at the time, I was doing other things without connection issues. 

Here's the full log just in case.

![photo_2022-04-16_14-25-03](https://user-images.githubusercontent.com/72584905/163687114-40a0f075-5e57-4213-aecf-9f3d92569375.jpg)

**Comment by rtoip** [comment]
> Even if you try to sync again? 

Just updated an iPhone 11 to from iOS 15.3.1 to iOS 15.4.1.  ==> sync still does not work.

**Comment by sif** [comment]
@rtoip Did you update Joplin too?

**Comment by clintconklin** [comment]
Fixed here with ios 15.4.1 and joplin 12.7.2... thanks @laurent22 !

**Comment by cjdaniel** [comment]
All fixed here: iOS 15.4.1 and Joplin 12.7.2. Thank you!!

**Comment by Simmiah** [comment]
Last night I updated my iOS app to version 12.7.2 and I found the previous `TypeError: Network request failed` error resolved and I was able to update my iOS app.  However as of this evening the `TypeError: Network request failed` started again and I can no longer sync.

I'm on an old iPhone 5S running iOS 12.5.5 (most up to date iOS I can run).
Joplin 12.7.2

Screenshot of the log from my iOS app
![joplin_error](https://user-images.githubusercontent.com/4175038/163700187-4a3747c9-a98d-4f9f-83fa-34daf846ae18.jpg)

**Comment by sif** [comment]
It looks like you have airplane mode turned on in the screenshot.

**Comment by laurent22** [comment]
I can't replicate any Network Request Failed error anymore. I thought it could happen with an empty resource but doesn't look like it does.

**Comment by CarlosZBent** [comment]
It's been working without issues for me since yesterday. Didn't even need to upgrade from iOS 15.4 to 15.4.1

**Comment by rtoip** [comment]
> Just updated an iPhone 11 to from iOS 15.3.1 to iOS 15.4.1. ==> sync still does not work.

After updating my iPhones (13,11) to 15.4.1 and updating Joplin the sync function works again.

**Comment by Simmiah** [comment]
> It looks like you have airplane mode turned on in the screenshot.

I do but I'm on wifi just no cellular data to sync. I use this mode to not deal with the spam texts I get in the country I'm in.

**Comment by Simmiah** [comment]
> Last night I updated my iOS app to version 12.7.2 and I found the previous `TypeError: Network request failed` error resolved and I was able to update my iOS app. However as of this evening the `TypeError: Network request failed` started again and I can no longer sync.
> 
> I'm on an old iPhone 5S running iOS 12.5.5 (most up to date iOS I can run). Joplin 12.7.2

As of tonight the issue appears resolved for me running the same iOS and Joplin version. I was able to sync both from my desktop to my iOS app and vice versa. No errors in my log on my iOS app anymore.

**Comment by Simmiah** [comment]
> It looks like you have airplane mode turned on in the screenshot.

I put my phone on airplane mode but still run wifi to use apps. This is because the country's cell service I'm in for some reason spams me with texts. I haven't previously run into any issues using wifi this way for my apps so far and take my phone off airplane mode during the day.

**Comment by daknour** [comment]
Hi! I had experienced this issue in the last months with IOS 14.3.1 and previous versions of Joplin

I am experiencing it again in IOS 15.5 and Joplin 12.8.1.

I use Joplin on an iPad, an iPhone and a Windows PC and this is happening to me more and more frequently in the last months, but just in the iPad. I clarify that on the phone I don't usually create or modify notes, I just read them. Maybe it is why it never happens to me there. But I do update my notes on both the iPad and PC frequently.

When Joplin "freezes" and loses its ability to sync with Dropbox on the iPad, and if I manage to temporarily fix the problem, I then find several file conflicts that I am need to review in detail or I even lose part of the content of my notes.

On the PC I also keep the app updated to its latest version.

**Comment by daknour** [comment]
### More details

1. **About Joplin on IOS**

```
Joplin 12.8.1
Database v41
FTS enabled: 1
```

2. **Status**

```
Attachments
Not downloaded: 0
Downloading:0
Downloaded and decrypted: 3
Downloaded and encrypted: O
Created locally: 1
Error: 0

Sync status (synced items / total items)
Note: 471/474
```


3. **Log**

```
07-19T18:25:05: RevisionService: "maintenance: Done in 115ms"
07-19T18:25:05: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:25:05: "Synchronisation is already in progress. State: in_progress"
07-19T18:25:05: "Starting scheduled sync"
07-19T18:25:05: "Preparing scheduled sync"
07-19T18:25:05: "Running background sync on timer..
07-19T18:25:05: RevisionService: "maintenance: Service is enabled"
07-19T18:25:05: RevisionService:"maintenance:Starting..
07-19T18:20:21: RevisionService: "maintenance:Done in 113ms"
07-19T18:20:21: RevisionService:
"collectRevisions: Created revisions for O notes"
07-19T18:20:20: RevisionService: "maintenance: Service is enabled"
07-19T18:20:20: RevisionService:"maintenance:Starting..
07-19T18:19:34: RevisionService: "maintenance: Done in 112ms"
07-19T18:19:34: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:19:34: RevisionService: "maintenance: Service is enabled"
07-19T18:19:34: RevisionService:"maintenance:Starting.
07-19T18:18:56: RevisionService: "maintenance: Done in 116ms"
07-19T18:18:56: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:18:56: RevisionService: "maintenance: Service is enabled"
07-19T18:18:56: RevisionService:"maintenance:Starting.
07-19T18:17:48: RevisionService: "maintenance: Done in 135ms"
07-19T18:17:48: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:17:48: RevisionService: "maintenance: Service is enabled"
07-19T18:17:48: RevisionService:"maintenance:Starting.
07-19T18:16:57: RevisionService: "maintenance: Done in 109ms"
07-19T18:16:57: "ResourceService:deleteOrphanResources:""]"
07-19T18:16:57: "ResourceService:indexNoteResources:Completed"
07-19T18:16:57: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:16:57: RevisionService:"maintenance:Starting.
07-19T18:16:57: RevisionService: "maintenance: Service is enabled"
07-19T18:16:57: "ResourceService:indexNoteResources:Start"
07-19T18:16:51: "Synchronisation is already in progress. State: in _progress"
07-19T18:16:51: "Starting scheduled sync"
07-19T18:16:51: "Preparing scheduled sync"
07-19T18:16:41: "Synchronisation is already in progress. State: in_progress"
07-19T18:16:41: "Starting scheduled sync"
07-19T18:16:41: "Preparing scheduled sync"
07-19T18:16:24: "SearchEngine:UpdatingFTStable.
07-19T18:16:18: RevisionService: "maintenance: Done in 120ms"
07-19T18:16:18: RevisionService: "collectRevisions: Created revisions for O notes"
07-19T18:16:18: RevisionService:"maintenance:Serviceisenabled"
```

4. **Log (Errors only)**

```
07-06T18:53:44: Synchronizer: "Could not refresh lock - cancelling sync. Error was:". "Error: Lock has expired
Code:lockExpired
construct@[nativecode]
o@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:59:331
construct@(nativecode]
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.isbundle:986:486
u@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:986:734
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:1031:10686
p@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:77:588
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:77:1893
s@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:1031:987
u@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:71:157
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:71:866
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:1672
k@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:500
w@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:898
callReactNativeMicrotasks@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:3054
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:2868
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:960
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:2504
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:919
value@[native codel
value@[native code]"
07-06T18:49:04: Synchronizer: "There was some errors:"
07-06T18:49:04: Synchronizer: "Sync target is locked - aborting API call"
07-06T18:49:03: Synchronizer: "Could not refresh lock - cancelling sync. Error was:", "Error: Lock has expired
Code:lockExpired
construct@[native codel
o@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:59:331
construct@(nativecode]
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:986:486
u@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:986:734
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:1031:10686
p@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:77:588
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:77:1893
s@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:1031:987
u@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:71:157
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:71:866
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:1672
k@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:500
w@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:898
callReactNativeMicrotasks@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:79:3054
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:2868
@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:960
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.jsbundle:38:2504
value@/private/var/containers/Bundle/Application/CE9FD087-3910-4FB5-8476-FF095836956D/Joplin.app/main.isbundle:38:919
value@[native code]
value@(nativecode]"
```

**Comment by daknour** [comment]
Nobody else is having this issue?

**Comment by sif** [comment]
@daknour At the time of this writing, the latest version for iOS and iPadOS is 15.6. The latest Joplin on desktop is 2.8.8 and on mobile is 12.8.1. 

Just to confirm, you are only experiencing this issue on your iPad? Between your desktop and iPhone, you don't experience this at all?

Can you describe what you mean by freezes?

When you said that you managed to temporarily fix the problem, what was the fix?

Can you please try the following steps:
1. Make sure your internet is stable (update router, check firewall settings, etc.). I would do the following on cellular if possible. 
2. You may not have the ability to go cellular on your desktop devices but then if the desktop devices are the only ones experiencing this, we've just narrowed down the problem.
3. If you have any plugins, disable them before trying.
4. Ensure that you have the latest iPhone updates and Joplin on iPhone.
5. Hit sync just once on iPhone. Let it finish.
6. Do the same with your iPad. Check both iPad version and Joplin on iPad. Hit sync just once. 
7. Ensure that you have the latest desktop updates and Joplin on desktop.
8. Hit sync just once.

If everything works, feel free to enable the plugins again.

**Comment by daknour** [comment]
The temporary fix is to locate the differences manually. I follow the next process:

1. Configuration > Tools > Sync Status > Check the number of notes in each folder/notebook to compare between iPad and PC and locate where there may be different number of notes.
2. In each folder/notebook, I sort the notes by update date and then, I look for the most recently updated notes and manually compare them, reading both iPad and PC notes.
3. I copy the different parts in an alternative notes App.
4. I turn off the network and delete all the notes on the iPad, then I unistall Joplin, connect the network again, reinstall Joplin and configure it again.
5. After all the notes are created and synced again in the iPad, I update the differences I extracted to the alternative notes App.
6. All the notes sync again for a few days among all the devices. Then it breaks again, just in the iPad.
 
It is so tedious that I have stopped repeating this process. I am waiting for a new Joplin update with a fix and if it is not fixed I will consider other options.

My Internet connection is stable. I just use the iPad with WiFi and I have much, much more bandwidth than I have ever needed.

The iPhone may connect to the network over wifi or cellular, but it has never failed. This said, I should add that I never modify my Joplin notes in the phone, I just sync them and read them in the phone. The modifications I make are only in the iPad and Windows PC.

I do not use any plugins. My Joplin version is automatically updated on my IOS devices and I always keep an eye to the updates on the PC as well, even more since I have this problem. So the versions are always the most recent.

And I hit sync only once because that's the only way. Once you hit it, it stays trying to sync the whole time you have Joplin open, even in the background. I could try to cancel, but the app doesn't obey. 

I have never opened Joplin on more than one device at a time. When I use it on the iPad, the PC is off and if I open it on the PC, I have also closed Joplin on the iPad first. Same with the phone, if I open Joplin there it's because I have neither the PC nor the tablet handy and when I'm done using Joplin, I close the App. On the iPad I rarely have more than two or three Apps open at the same time as I close all Apps when not in use. So if Joplin is open while another app is open as well, it's because I'm taking notes from it, for example from web pages I visit with Safari. Lastly, at a minimum I restart my iPad and phone once or twice a week.

I have been experiencing this issue for the last few months. When it started I used IOS 14.3.1, though I do not remember which version of Joplin I had in April. Since then, I have been updating Joplin and IOS too. I am now experiencing it again with IOS 15.6 and Joplin 12.8.1.

I hope I have answered all the questions.
